script.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+# enumerable ruby module
+module Enumerable
+  def my_each
+    i = 0
+    length.times do
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    length.times do
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    i = 0
+    res_arr = []
+    length.times do
+      res_arr << self[i] if yield(self[i]) == true
+      i += 1
+    end
+    res_arr
+  end
+
+  def my_all?
+    i = 0
+    length.times do
+      return false if yield(self[i]) == false
+
+      i += 1
+    end
+    true
+  end
+
+  def my_none?
+    i = 0
+    length.times do
+      return false if yield(self[i]) == true
+
+      i += 1
+    end
+    true
+  end
+
```

Comment:
In `my_none?`
-  my_none? when no block or argument is given returns true only if none of the collection members is true
- my_none? when a class is passed as an argument returns true if none of the collection is 
a member of such class
- my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
-  my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern